### PR TITLE
fix: instrumentation of Rails 7

### PR DESCRIPTION
**Pull request by jimeh**
ActionView instrumentation no longer fails by having it use the same
subscriber logic as used for Rails 6.

Also add Rails 7 Appraisals to all relevant gems which has any Rails
Appraisals.

### CODE REVIEW

File: instrumentation/action_view/lib/opentelemetry/instrumentation/action_view/span_subscriber.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ module ActionView
       def self.subscribe(pattern = nil, callable = nil)
         ActiveSupport::Notifications.subscribe(pattern, callable)
         ::ActiveSupport::Notifications.notifier.synchronize do
-          if ::Rails::VERSION::MAJOR == 6
+          if ::Rails::VERSION::MAJOR >= 6
```
[the remark this answers is not in the document]
Comment (jimeh) [reply instrumentation/action_view/lib/opentelemetry/instrumentation/action_view/span_subscriber.rb:20]: Oh, nice. However that PR was merged just over 3 weeks after the latest alpha was published, so it's definitely not included right now.

[the remark this answers is not in the document]
Comment (SomalianIvan) [reply instrumentation/action_view/lib/opentelemetry/instrumentation/action_view/span_subscriber.rb:20]: 👍

**Review by ahayworth — DISMISSED** [review]
This looks fine to me - once https://github.com/rails/rails/pull/43282 is in a released version, we can definitely consider dropping our subscription-ordering hacks for Rails 7 and beyond. 😄

**Comment by ahayworth** [comment]
@jimeh I approved the Actions workflow runs for you - thank you for being a first time contributor! - but now the tests seem to be failing.

**Comment by jimeh** [comment]
@ahayworth Thanks, and the test failures are rather annoying. I had it all passing locally.

I'll try and resolve the failures a bit later today when I'm free :)

**Comment by jimeh** [comment]
Test failures were due to Rails 7.x requiring Ruby 2.7.0 or later. I've just pushed a fix (hopefully), and also rebased the PR on top of latest state of the `main` branch.

**Review by ahayworth — APPROVED** [review]
Awesome, looks good - thanks!

**Pull request merged**